src/parser/postgresparser.cpp

```diff
@@ -800,6 +800,143 @@ parser::SQLStatement* PostgresParser::CreateTransform(CreateStmt* root) {
+  char* name = (reinterpret_cast<value*>(type_name->names->tail->data.ptr_value)
+                    ->val.str);
+  parser::FuncParameter* result = nullptr;
+
+  // Transform parameter type
+  if ((strcmp(name, "int") == 0) || (strcmp(name, "int4") == 0)) {
+    data_type = FuncParameter::DataType::INT;
+  } 
+  //*******See which of the below are needed later***********
+  // I think text/varchar may be used for the pg_catalog
+  else if (strcmp(name, "varchar") == 0) {
+    data_type = FuncParameter::DataType::VARCHAR;
+  } else if (strcmp(name, "int8") == 0) {
+    data_type = FuncParameter::DataType::BIGINT;
+  } else if (strcmp(name, "int2") == 0) {
+    data_type = FuncParameter::DataType::SMALLINT;
+  } else if (strcmp(name, "text") == 0) {
+    data_type = FuncParameter::DataType::TEXT;
+  } else if (strcmp(name, "tinyint") == 0) {
+    data_type = FuncParameter::DataType::TINYINT;
+  } else if(strcmp(name, "bool") == 0) {
+     data_type = FuncParameter::DataType::BOOL;
+  } else {
+    LOG_ERROR("Column DataType %s not supported yet...\n", name);
+    throw NotImplementedException("...");
+  }
+
+  // Transform Varchar parameter name
+  std::string param_name(root->name ? root->name : "");
+  result = new FuncParameter(param_name, data_type);
+
+  return result;
+}
+
+// This helper function takes in a Postgres TypeName object and transforms
+// it into a Peloton ReturnType object
+parser::ReturnType* PostgresParser::ReturnTypeTransform(TypeName* root) {
+  parser::ReturnType::DataType data_type;
+  char* name = (reinterpret_cast<value*>(root->names->tail->data.ptr_value)
+                    ->val.str);
+  parser::ReturnType* result = nullptr;
+
+  // Transform parameter type
+  if ((strcmp(name, "int") == 0) || (strcmp(name, "int4") == 0)) {
+    data_type = FuncParameter::DataType::INT;
+  } 
+  //*******See which of the below are needed later***********
+  // I think text/varchar may be used for the pg_catalog
+  else if (strcmp(name, "varchar") == 0) {
+    data_type = FuncParameter::DataType::VARCHAR;
+  } else if (strcmp(name, "int8") == 0) {
+    data_type = FuncParameter::DataType::BIGINT;
+  } else if (strcmp(name, "int2") == 0) {
+    data_type = FuncParameter::DataType::SMALLINT;
+  } else if (strcmp(name, "text") == 0) {
+    data_type = FuncParameter::DataType::TEXT;
+  } else if (strcmp(name, "tinyint") == 0) {
+    data_type = FuncParameter::DataType::TINYINT;
+  } else if(strcmp(name, "bool") == 0) {
+     data_type = FuncParameter::DataType::BOOL;
+  } else {
+    LOG_ERROR("Column DataType %s not supported yet...\n", name);
+    throw NotImplementedException("...");
+  }
+
+  // Transform Varchar parameter name
+  result = new ReturnType(data_type);
+
+  return result;
+}
+
+// This function takes in a Postgres CreateFunctionStmt parsenode
+// and transfers into a Peloton CreateFunctionStatement parsenode.
+// Please refer to parser/parsenode.h for the definition of
+// CreateFunctionStmt parsenodes.
```

Comment:
The file is "parser/parsenodes.h" (with an 's'), but that's a minor quibble with an otherwise useful comment.